src/ORM/Hierarchy/Hierarchy.php

```diff
@@ -303,17 +303,16 @@ public function duplicateWithChildren(): DataObject
     {
         $owner = $this->getOwner();
         $clone = $owner->duplicate();
-        $children = $owner->AllChildren();
+        // Disabling sort improves performance
+        $children = $owner->AllChildren()->sort(null);
         $sortField = $owner->getSortField();
 
-        $sort = 1;
         foreach ($children as $child) {
             $childClone = $child->duplicateWithChildren();
             $childClone->ParentID = $clone->ID;
             if ($sortField) {
                 //retain sort order by manually setting sort values
-                $childClone->$sortField = $sort;
-                $sort++;
+                $childClone->$sortField = $child->$sortField;
```

Comment:
This was originally added in https://github.com/silverstripe/silverstripe-cms/pull/1636
I _don't think_ it's really needed, but since there's no indication of what caused the original bug we should keep this code that manually sets the sort field.

We can assume the original children had unique sort values (the original bug is that the _duplicates_ ended up with identical sort values to one another) so just setting the clone sort value to the original child's sort value should be sufficient.

Reply:
This shouldn't be in this PR, it's changing logic which is unrelated to the performance enhancement

Reply:
It's explicitly _keeping_ the bugfix. Before that bug fix no sort was being set, so the values were all identical. As per the description in the PR I linked to above:

> This fixes an issue where pages duplicated with their children cause the new child pages to all have the same Sort value causing the child pages to jump around on save.

I have updated this to _still give values which are not all the same_ which means the bug fix is preserved. Note that the test which was added in that same PR is still passing.

Reply:
It's going to have a different result than before though

Before

```
PageA
- ChildAA.Sort = 3
- ChildAB.Sort = 7

Duplicate PageA to PageB

PageB
- ChildBA.Sort = 1
- ChildBB.Sort = 2
```

After

```
PageA
- ChildAA.Sort = 3
- ChildAB.Sort = 7

Duplicate PageA to PageB

PageB
- ChildBA.Sort = 3
- ChildBB.Sort = 7
```

You could argue that the new logic is an improvement, though it's still clearly a change and is outside the scope of a performance enhancement where things should behave the same way before and after